KitePrintSDK/src/main/java/ly/kite/analytics/Analytics.java

```diff
@@ -0,0 +1,660 @@
+      }
+
+    return ( mCachedUniqueUserId );
+    }
+
+
+  /*****************************************************
+   *
+   * Returns a JSON object containing the standard properties
+   * that we upload with every event.
+   *
+   * The same properties are uploaded with every analytics
+   * event, so it makes sense to cache them. However, we don't
+   * want to return the actual cached properties object, as
+   * every time we add more properties, they leak into all the
+   * events.
+   *
+   * So we create the properties as a map, and then copy them
+   * into a new JSON object.
+   *
+   *****************************************************/
+  private JSONObject getPropertiesJSONObject()
+    {
+    // If we haven't already create the properties object - do
+    // so now.
+
+    if ( mCachedPropertiesMap == null )
+      {
+      mCachedPropertiesMap = new HashMap<>();
+
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_API_TOKEN,        MixpanelAgent.API_TOKEN );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_UNIQUE_USER_ID,   getUniqueUserId() );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_APP_ID,           mContext.getPackageName() );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_APP_NAME,         mContext.getString( mContext.getApplicationInfo().labelRes ) );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_APP_VERSION,      BuildConfig.VERSION_NAME );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_PLATFORM,         PLATFORM_JSON_PROPERTY_VALUE_ANDROID );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_PLATFORM_VERSION, Build.VERSION.RELEASE );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_MODEL,            Build.MODEL );
+
+
+      DisplayMetrics displayMetrics = mContext.getResources().getDisplayMetrics();
+
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_SCREEN_WIDTH,     displayMetrics.widthPixels );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_SCREEN_HEIGHT,    displayMetrics.heightPixels );
+
+
+      KiteSDK             kiteSDK     = KiteSDK.getInstance( mContext );
+      KiteSDK.Environment environment = kiteSDK.getEnvironment();
+
+      String environmentString;
+
+      switch ( environment )
+        {
+        case LIVE:    environmentString = ENVIRONMENT_JSON_PROPERTY_VALUE_LIVE;    break;
+        case TEST:    environmentString = ENVIRONMENT_JSON_PROPERTY_VALUE_TESTING; break;
+        case STAGING: environmentString = ENVIRONMENT_JSON_PROPERTY_VALUE_STAGING; break;
+        default:      environmentString = ENVIRONMENT_JSON_PROPERTY_VALUE_UNKNOWN; break;
+        }
+
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_ENVIRONMENT,      environmentString );
+
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_API_KEY,          kiteSDK.getAPIKey() );
+
+
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_KITE_SDK_VERSION, KiteSDK.SDK_VERSION );
+      mCachedPropertiesMap.put( JSON_PROPERTY_NAME_LOCALE_COUNTRY,   Locale.getDefault().getCountry() );
```

Comment:
Could you move to `Country.displayName()`? -- that way we'll have consistent names with iOS and comparison is a little more straight forward. Also we won't need to lookup names of obscure country codes.